authority/provisioner/collection.go

```diff
@@ -91,10 +92,16 @@ func (c *Collection) LoadByToken(token *jose.JSONWebToken, claims *jose.Claims)
 	if err := token.UnsafeClaimsWithoutVerification(&payload); err != nil {
 		return nil, false
 	}
-	// Audience is required
-	if len(payload.Audience) == 0 {
+	if len(payload.ServiceAccountName) > 0 {
+		if p, ok := c.Load("k8ssa/" + K8sSAName); ok {
```

Comment:
Looks like we are loading the provisioner `k8ssa/k8sSA-default` but storing `k8ssa/<provisioner.Name>`.

Reply:
Yea, so the provisioner.name get's hardcoded from the cli. I can leave some documentation in `provisioner add --type k8ssa` to that effect. And also do some validation in the provisioner.init to prevent the CA from starting up if the name doesn't match the default.

Reply:
I've changed this so we are overwriting the provisioner name with the hardcoded one. And added validation in the CA to fail startup if a K8sSA provisioner has the wrong name (or if there's more than one such provisioner).

Reply:
I would prefer to let the user set a custom name, and just use the fixed `GetID()`, this ID is used in `Collection.Store()`.

Reply:
And by the way, `LoadByCertificate` does not have support for K8sSA.